### PR TITLE
feat: Add well-known log selectors to console output

### LINKED ISSUE #156: FEAT: Support additional console log payloads
#### 📝 Description

era-test-node currently seems to only support `console.log` that is limited to specific types. If you try to make use of `console.logBytes(value1);` the output in the logs is as follows: 

Screenshot 2023-10-03 at 9 06 46 AM

It would be good for users to be able to log any type according to hardhat console log [here](https://github.com/NomicFoundation/hardhat/blob/main/packages/hardhat-core/console.sol).

#### 🤔 Rationale

If I want to log bytes its easier to make use of `console.logBytes(value1);`.

### PR DESCRIPTION
**Pull request by nbaztec**
# What :computer: 
* Adds well known log selectors to console for pretty printing

# Why :hand:
* Helpful to view output logs, e.g. `logBytes1(...)` which wasn't supported before

# Evidence :camera:
```sol
console.logBytes1(bytes1(0x01));
```
image

# Notes :memo:
* Fixes #156

### CODE REVIEW

File: src/console_log.rs [hunk 1]
```diff
@@ -71,3 +62,432 @@ impl ConsoleLogHandler {
...
+                state_mutability: StateMutability::View,
+            }
+        })
+        .collect()
+}
+
+/// Represents the common log functions.
+/// See https://github.com/NomicFoundation/hardhat/blob/main/packages/hardhat-core/console.sol
+const LOG_FUNCTIONS: [&str; 378] = [
+    "log(int256)",
```
Comment (nbaztec) [inline src/console_log.rs:105]: I can also dynamically generate this list, though unsure which is better.

```rust
        let mut func_decl = IndexSet::<String>::default();
        func_decl.extend([
            String::from("log(int256)"),
            String::from("log(uint256)"),
            String::from("log(string)"),
            String::from("log(bool)"),
            String::from("log(address)"),
            String::from("log(bytes)"),
        ]);

        // bytes1 - bytes32
        func_decl.extend(
            (1..=32)
                .into_iter()
                .map(|i| format!("log(bytes{i})"))
                .collect_vec(),
        );

        // combination of ("uint256","string", "bool", "address") upto 4 params
        for i in 1..=4 {
            func_decl.extend(
                ["uint256", "string", "bool", "address"]
                    .iter()
                    .cloned()
                    .combinations_with_replacement(i)
                    .map(|c| {
                        c.into_iter()
                            .permutations(i)
                            .map(|c| format!("log({})", c.join(",")))
                    })
                    .flatten()
                    .collect_vec(),
            );
        }
```

File: src/console_log.rs [hunk 2]
```diff
@@ -24,20 +24,11 @@ pub const CONSOLE_ADDRESS: &str = "0x000000000000000000636f6e736f6c652e6c6f67";
 
 impl Default for ConsoleLogHandler {
     fn default() -> Self {
-        let mut val: Value =
-            serde_json::from_slice(include_bytes!("deps/contracts/Console.json")).unwrap();
```
Comment (MexicanAce) [inline src/console_log.rs:28]: We can delete the `Console.sol` and corresponding files with this update 👍

File: src/console_log.rs (hunk 1 above)
Comment (MexicanAce) [reply src/console_log.rs:105]: I'm fine either way tbh. I'm leaning towards the explicit list as I don't think `Console.sol` is expected to change from Hardhat, and it provides a clear list of what's supported.

Comment (nbaztec) [reply src/console_log.rs:105]: Yes, I'm also leaning towards a hardcoded list.

**Review by MexicanAce — APPROVED** [review]
LGTM

**Pull request merged**